source/base/process_grid.cc

```diff
@@ -170,9 +170,8 @@ namespace Utilities
 
       // Create the communicator based on the group
       // Note that on most cores the communicator will be MPI_COMM_NULL.
-      // FIXME: switch to MPI_Comm_create_group for MPI-3 so that only processes within the to-be subgroup call this
-      ierr = MPI_Comm_create(mpi_communicator, inactive_with_root_group,
-                             &mpi_communicator_inactive_with_root);
+      ierr = Utilities::MPI::create_group(mpi_communicator, inactive_with_root_group,
+                                          55, &mpi_communicator_inactive_with_root);
```

Comment:
I'm not sure this is correct. `MPI_Comm_create_group` is collective only on those processes that are actually included in the group, i.e., exactly those that are listed in `inactive_with_root_group`. But the function here is called on *all* processes that are active in `mpi_communicator`, which is a superset if I understand this correctly.

Don't we have to guard this call with an `if` statement then? Note that this is different from `MPI_Comm_create`, which is collective with all processes that are part of the communicator, not just the subgroup.

Reply:
> Don't we have to guard this call with an if statement then? Note that this is different from MPI_Comm_create, which is collective with all processes that are part of the communicator, not just the subgroup.

No, we don't need to guard this. Citing the [MPI-3.1 standard](https://www.mpi-forum.org/docs/mpi-3.1/mpi31-report.pdf), p. 243: 

>[...] If  the calling process is not a member of `group`, e.g., `group` is `MPI_GROUP_EMPTY`, 
then the call is a local operation and `MPI_COMM_NULL` is returned as `newcomm`.